File: go2shop-app/src/app/pages/order/order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpResponse } from '@angular/common/http';
import { Observable } from 'rxjs';
import { IOrder, IOrderSearchDTO } from './order.model';

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  constructor(private http: HttpClient) { }

  private api = 'api/orderService/order';

  placeOrder(order: IOrder): Observable<HttpResponse<IOrder[]>> {
    return this.http.post<IOrder[]>(`${this.api}/createOrder`, order, { observe: 'response' });
  }

  getOrder(orderId: number): Observable<HttpResponse<IOrder>> {
    return this.http.get<IOrder>(`${this.api}/order/${orderId}`, { observe: 'response' });
  }

  cancelPayment(orderIds: number[]): Observable<HttpResponse<void>> {
    return this.http.post<void>(`${this.api}/cancelPayment`, orderIds, { observe: 'response' });
  }

  confirmPayment(payOption: string, orderIds: number[]): Observable<HttpResponse<void>> {
    return this.http.post<void>(`${this.api}/confirmPayment/${payOption}`, orderIds, { observe: 'response' });
  }

  searchOrders(search: IOrderSearchDTO): Observable<HttpResponse<IOrder[]>> {
    return this.http.post<IOrder[]>(`${this.api}/order/search`, search, { observe: 'response' });
  }

  orderReceived(order: IOrder): Observable<HttpResponse<void>> {
    return this.http.post<void>(`${this.api}/orderReceived`, order, { observe: 'response' });
  }

  confirmDelivery(order: IOrder): Observable<HttpResponse<void>> {
    return this.http.post<void>(`${this.api}/confirmDelivery`, order, { observe: 'response' });
  }
}

File: go2shop-order-service/src/main/java/com/go2shop/order/service/OrderService.java
package com.go2shop.order.service;

import java.util.List;
import java.util.Optional;

import com.go2shop.common.exception.BusinessException;
import com.go2shop.model.order.CreateOrderDTO;
import com.go2shop.model.order.OrderDTO;
import com.go2shop.model.order.OrderSearchDTO;

public interface OrderService {
	List<OrderDTO> createOrder(CreateOrderDTO order) throws BusinessException;
	Optional<OrderDTO> findOrderById(Long id);
	void cancelPayment(List<Long> orderIds);
	void confirmPayment(String payOption, List<Long> orderIds) throws BusinessException;
	List<OrderDTO> searchOrders(OrderSearchDTO searchDTO);
	void orderReceived(OrderDTO order);
	void confirmDelivery(OrderDTO order);
}
